core/src/main/java/io/github/tcdl/msb/collector/Collector.java

```diff
@@ -42,7 +41,8 @@
 
     private int timeoutMs;
     private int currentTimeoutMs;
-    private long waitForAcksUntil;
+    private Integer waitForAcksMs;
+    private Long waitForAcksUntil;
```

Comment:
Essentially this variable represents some moment of time. Maybe it's more natural and expressive to use `java.time.Instant` instead of plain `java.lang.Long`?